src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -420,6 +427,29 @@ export class StudyViewPageStore
         );
     }
 
+    private setCustomChart(chartMeta: ChartMetaWithDimensionAndChartType) {
+        const uniqueKey = chartMeta.uniqueKey;
+        // initialize chartsType.get(uniqueKey) if not already set
+        if (!this.chartsType.has(uniqueKey)) {
+            this.chartsType.set(uniqueKey, chartMeta.chartType);
```

Comment:
i know this is legacy but should probably be this.chartTypes or something even more descriptive.  if it can be easily improved lets do it